### PR TITLE
Updated breadcrumbs logic

### PR DESCRIPTION
**Pull request by ghost**
Changed harvest_course_tree_new to work with a list of dicts as output instead of a dict which enables easier work with pandas.

Created a logic whereby module, section, lesson, unit and component are only populated if applicable (e.g. if the block is a module, the rest of the fields will be None)

Created function like dict.get() which safely extracts an index from a list and returns None if the index is out of bounds.

### CODE REVIEW

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py [hunk 1]
```diff
@@ -30,8 +30,16 @@
 
 BREADCRUMB_INDEX_URL = settings.LMS_SYLLABUS
 
+BLOCK_TYPES = {
```
Comment (yoniLavi) [inline lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py:34]: Please add a comment to explain the semantics here. From what I understand, the keys are internal edx terminology and the values are our terminology (also based on edx), right?

Also, please add a trailing comma on line 37, it makes diffs nicer

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py [hunk 2]
```diff
@@ -40,25 +48,58 @@ def harvest_course_tree(tree, output_dict, prefix=()):
     block_name = tree.display_name
     block_breadcrumbs = prefix + (tree.display_name,)
     block_id = tree.location.block_id
-
-    output_dict[block_id] = block_breadcrumbs
+    category = (tree.category,)
```
Comment (yoniLavi) [inline lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py]: I feel uneasy about a singular variable name for a collection (even if it has just one item. I'd just add the category directly on the line below, i.e.:
`output_dict[block_id] = block_breadcrumbs + (tree.category,)`

Comment (yoniLavi) [reply lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py]: Also, please add a comment to explain the purpose of this

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py [hunk 3]
```diff
@@ -40,25 +48,58 @@ def harvest_course_tree(tree, output_dict, prefix=()):
...
+
+    children = tree.get_children()
+    for subtree in children:
+        harvest_course_tree_new(subtree, output_list, prefix=block_breadcrumbs)
+    
+
 def harvest_program(program):
     """Harvest the breadcrumbs from all components in the program
 
     Returns a dictionary mapping block IDs to the matching breadcrumbs
```
Comment (yoniLavi) [inline lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py]: Having switched to a list, you should update the docstring here

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py [hunk 4]
```diff
@@ -69,11 +110,11 @@ def get_breadcrumb_index(URL):
     df_breadcrumb_idx = df_breadcrumb_idx.T
     df_breadcrumb_idx = df_breadcrumb_idx.reset_index()
     df_breadcrumb_idx.rename(columns={'index':'order_index'}, inplace=True)
-
     # TODO: remove following line, once the [beta] suffix is removed in the LMS
-    df_breadcrumb_idx['module'] = df_breadcrumb_idx['module'].replace('Careers','Careers [Beta]')
+    df_breadcrumb_idx['module'] = df_breadcrumb_idx['module'].replace(
+        'Careers','Careers [Beta]')
```
Comment (yoniLavi) [inline lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py]: Missing space after comma, both here and on line 117

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py [hunk 5]
```diff
@@ -40,25 +48,58 @@ def harvest_course_tree(tree, output_dict, prefix=()):
...
-    output_dict[block_id] = block_breadcrumbs
+    category = (tree.category,)
+    output_dict[block_id] = block_breadcrumbs + category
 
     children = tree.get_children()
     for subtree in children:
         harvest_course_tree(subtree, output_dict, prefix=block_breadcrumbs)
 
 
+def get_safely(breadcrumbs, index):
```
Comment (yoniLavi) [inline lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py:44]: I appreciate this :)

**Review by yoniLavi — APPROVED** [review]

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py (hunk 1 above)
Comment (ghost) [reply lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py:34]: Added the trailing comma and some comments to clarify the BLOCK_TYPES dict keys and values.

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py (hunk 4 above)
Comment (ghost) [reply lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py]: Added space

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py (hunk 3 above)
Comment (ghost) [reply lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py]: Updated the docstring

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py (hunk 5 above)
Comment (ghost) [reply lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py:44]: Thanks :)

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py (hunk 2 above)
Comment (ghost) [reply lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py]: Ah, this was actually part of the old and unused function and used for testing that I had forgot to remove. I removed the unused function now and changed the name of the new function from harvest_course_tree_new to harvest_course_tree. Also, tested that this is still working after and it is.

**Pull request merged**